### ISSUE TITLE
DB::Exception: Illegal column for DataTypeNullable error.

### ISSUE THREAD

**Issue by prasadkris** [getsentry/self-hosted#2312]
### Problem Statement

Greetings! 👋 We have a Sentry deployment on our self-hosted Kubernetes cluster (k8s version: v1.25.7) which was deployed using the Sentry-Kubernetes Helm chart, version `15.x.x.` This serves Sentry version 22.7.0, and it worked fine without any issues. 

Today, I attempted an upgrade to a new version by following the [suggested](https://github.com/sentry-kubernetes/charts) Helm chart upgrade path (`15.x.x -> 16.x.x -> 17.x.x -> 18.x.x`). The upgrade went smoothly without any issues.

However, after the upgrade, a **couple of our projects** are having problems processing events. When I go to the issues page, I see an error screen with "unable to fetch project details" message as shown in the below screenshot image, and I am unable to access the general settings page for these projects.

<img width="1198" alt="unable" src="https://github.com/getsentry/self-hosted/assets/97734766/f9ce340e-3303-42e1-9516-207a50c8c80c">

<img width="1204" alt="image2" src="https://github.com/getsentry/self-hosted/assets/97734766/c4e16b40-ab1f-437e-99b6-b9986a177087">

This probelm only exists with a ciuple of the projects andf I see the following exception in the `sentry-web` and the `snuba-api` pod logs when I navigate to to the "problematic" projects: 

```
QueryExecutionError)(
sentry.utils.snuba.QueryExecutionError: DB::Exception: Illegal column for DataTypeNullable
```

The `sentry-web` pod logs are available [here](https://gist.github.com/prasadkris/4bdd68a91dd455f9cb38bb6b624d2b80) and the `snuba-api` logs are available [here](https://gist.github.com/prasadkris/94a1830f23cca46a4929ab3925e7e024)

Any help or pointers on fixing this would be much appreciated. Please let me know if you need any further details. Thanks! 🙏

**Comment by hubertdeng123** [comment]
I am not familiar with helm-charts, but this may have something to do with a corrupt snuba migration. You'll have to take this up with `sentry-kubernetes` for more info

**Comment by prasadkris** [comment]
Thanks a lot for your response, @hubertdeng123. I performed an upgrade from chart v18 to v19.5.1 and I am able to view the issue details afterwards, but I am still unable to view the project details and the settings page for a few projects. Only 2 of our projects are exhibiting these problems out of the total 14 projects we have in Sentry.

<img width="1440" alt="Screenshot 2023-08-01 at 6 23 19 AM" src="https://github.com/getsentry/self-hosted/assets/97734766/6308e341-581a-4239-96e6-6047f3384e8b">

<img width="1411" alt="2" src="https://github.com/getsentry/self-hosted/assets/97734766/5717baa4-c017-445f-ab40-4ddaf58a4a6c">


I get the following exception in the `sentry-web` and `snuba-api` pods when this happens as explained in my initial post ([sentry-web](https://gist.github.com/prasadkris/4bdd68a91dd455f9cb38bb6b624d2b80), [snuba-api](https://gist.github.com/prasadkris/94a1830f23cca46a4929ab3925e7e024))

```
QueryExecutionError)(
sentry.utils.snuba.QueryExecutionError: DB::Exception: Illegal column for DataTypeNullable
```

Are we sure this is an issue with the snuba migration? I went through the snuba-migration job pod logs and couldn't find anything suspicious there. I have already opened https://github.com/sentry-kubernetes/charts/issues/946 for this and I have updated it with further details. Let's see how it goes, thanks again for your response 🙏

**Comment by hubertdeng123** [comment]
Hmm, this may be some issue then with some corrupt data in some column in clickhouse then.

**Comment by prasadkris** [comment]
Thanks @hubertdeng123 🙏 I have investigated this further by provisioning a brand new test setup and I can confirm that this is a problem with the upgrade from chart version 17.x.x to 18.x.x (sentry version `22.x.x` to `23.x.x`). I have updated the details at https://github.com/sentry-kubernetes/charts/issues/946#issuecomment-1659728151. Hopefully, we will get some update from the `sentry-kubernetes` team soon, thanks!
